### ISSUE TITLE
a referring WorkflowTemplate as Workflow and using WorkflowTemplates's entrypoint and Workflow Arguments

### ISSUE THREAD

**Issue by VictoryAnn** [argoproj/argo#5041]
## Summary 

```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: workflow-template-hello-world-
spec:
  workflowTemplateRef:
    name: workflow-template-submittable
```
I create the workflow, but get error that ```Failed to submit workflow: spec.entrypoint is required```

## Diagnostics

What Kubernetes provider are you using? 

Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T13:41:02Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T13:32:58Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}

What version of Argo Workflows are you running? 

workflow-controller : argoproj/workflow-controller:v2.11.6

argo: v2.12.6
  BuildDate: 2021-01-25T20:03:12Z
  GitCommit: 4cb5b7eb807573e167f3429fb5fc8bf5ade0685d
  GitTreeState: clean
  GitTag: v2.12.6
  GoVersion: go1.13
  Compiler: gc
  Platform: linux/amd64

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: workflow-template-hello-world-
spec:
  workflowTemplateRef:
    name: workflow-template-submittable
```

```
Paste the logs from the workflow controller:
kubectl logs -n argo $(kubectl get pods -l app=workflow-controller -n argo -o name) | grep ${workflow}

Name:                workflow-template-hello-world-p2spk
Namespace:           argo
ServiceAccount:      default
Status:              Failed
Message:             invalid spec: spec.entrypoint is required
Conditions:          
 Completed           True
Created:             Mon Feb 08 16:52:17 +0800 (13 seconds ago)
Started:             Mon Feb 08 16:52:17 +0800 (13 seconds ago)
Finished:            Mon Feb 08 16:52:17 +0800 (13 seconds ago)
Duration:            0 seconds
Progress:            

```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by sarabala1979** [comment]
@VictoryAnn  You can pass entrypoint with ‘—entry point’ option’

argo submit  <wf.yaml> —entrypoint main

**Comment by dan-j** [comment]
Apologies for reviving this issue, but I've also ran into the same problem on v3.2.0-rc1. Although my use-case may be different. I have a `WorkflowTemplate` which uses `workflowTemplateRef` to a `ClusterWorkflowTemplate`, and then I'm trying to submit a workflow from the `WorkflowTemplate`.

I've tried submitting from the UI and like `argo submit --from workflowtemplate/my-template` both with and without the `--entrypoint main` argument and I get the error:

```
Failed to submit workflow: template reference my-template.main not found
```
